Adding `BayesianGCNConv` for Bayesian Graph Neural Network implementation

gabyweg76 opened the pull request:
Trial of a Bayesian Convolution Layer inspired from 
`https://github.com/IntelLabs/bayesian-torch/blob/main/bayesian_torch/layers/variational_layers/linear_variational.py`
applied for GCNConv, the final step is to implement this conv layer in a generic graph neural network like a GraphUnet to create a Bayesian Graph Neural Network 

To cite their code : 

@software{krishnan2022bayesiantorch,
  author       = {Ranganath Krishnan and Pi Esposito and Mahesh Subedar},               
  title        = {Bayesian-Torch: Bayesian neural network layers for uncertainty estimation},
  month        = jan,
  year         = 2022,
  doi          = {10.5281/zenodo.5908307},
  url          = {https://doi.org/10.5281/zenodo.5908307}
  howpublished = {\url{https://github.com/IntelLabs/bayesian-torch}}
}

gabyweg76 committed:
Trial of Bayesian GCNConv implementation 

proposition needed to be  checked  and confirmed for public using (perspective implement this new GCNConv in a Bayesian Graph U-Net)

gabyweg76 committed:
Merge pull request #1 from gabyweg76/gabyweg76-patch-1

Trial of Bayesian GCNConv implementation

gabyweg76 committed:
BGCNConv corrected

gabyweg76 committed:
Merge pull request #2 from gabyweg76/gabyweg76-patch-2

BGCNConv corrected

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 reviewed (commented):
Correct some typos

gabyweg76 committed:
Update Bayesian_GCNConv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update Bayesian_GCNConv.py

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,228 @@
+    _cached_adj_t: Optional[SparseTensor]
+
+    def __init__(self, in_channels: int, out_channels: int, prior_mean=0,
+                 prior_variance=1, posterior_mu_init=0,
```
type hints

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,228 @@
+# layers/base_variational_layer.py
+
+
+def gcn_norm(edge_index, edge_weight=None, num_nodes=None, dtype=None):
```
Use `torch_geometric.nn.conv.gcn_conv.gcn_norm` instead.

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,228 @@
+        self._cached_edge_index = None
+        self._cached_adj_t = None
+
+    def kl_div(self, mu_q, sigma_q, mu_p, sigma_p):
```
type hints

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,228 @@
+        return kl
+
+    def forward(self, x: Tensor, edge_index: Adj,
+                edge_weight: OptTensor = None, return_kl=True) -> Tensor:
```
```suggestion
                edge_weight: OptTensor = None, return_kl: bool =True) -> Tensor:
```
type hints

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,228 @@
+        return edge_index, deg_inv_sqrt[row] * edge_weight * deg_inv_sqrt[col]
+
+
+class BGCNConv(MessagePassing):
```
Please also add some tests for `BGCNConv`.

gabyweg76 replied:
With pleasure, what kind of test are you thinking about ? We can use a graph from pyg datasets and test it with this new conv layer ?

EdisonLeeeee replied:
Just test the basic functionality of `BGCNConv`, you may refer https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/conv/test_sg_conv.py for a simple example :)

gabyweg76 replied:
ok will do that tomorrow morning haha, I am so sleepy rn

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,228 @@
+        self.cached = cached
+        self.normalize = normalize
+
+        self._cached_edge_index = None
```
```suggestion
```

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,228 @@
+        self.normalize = normalize
+
+        self._cached_edge_index = None
+        self._cached_adj_t = None
```
```suggestion
```

EdisonLeeeee reviewed (commented):
Thanks for adding this. Left some initial reviews.

gabyweg76 committed:
Update torch_geometric/nn/conv/Bayesian_GCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Update torch_geometric/nn/conv/Bayesian_GCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Update torch_geometric/nn/conv/Bayesian_GCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Incoming test functions coming soon

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Added first bunge of tests

I get those tests from `torch_geometric.test.nn.conv.tes_gcn_conv.py`. These are simple tests, let me know if we want to go further in details. Just  a reminder that the main theorical core of the conv layer is inspired from <https://github.com/IntelLabs/bayesian-torch>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 commented:
I added the tests I saw on `test.nn.conv.test_gcn_conv.py`, let me know if we have to go further in details. The basic theory can be found here `https://github.com/IntelLabs/bayesian-torch`

gabyweg76 committed:
Update Bayesian_GCNConv.py

gabyweg76 committed:
Update Bayesian_GCNConv.py

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,267 @@
+        return spmm(adj_t, x, reduce=self.aggr)
+
+
+def test_gcn_conv_with_sparse_input_feature(device="cpu"):
```
Let's move the tests to the `test/` folder.

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,267 @@
+        assert jit(x, edge_index, value).tolist() == out2.tolist()
+
+
+if __name__ == "__main__":
```
We don't need these anyway.

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,267 @@
+# layers/base_variational_layer.py
+
+
+class BayesianGCNConv(MessagePassing):
```
Please also add some doc-string.

EdisonLeeeee commented on torch_geometric/nn/conv/Bayesian_GCNConv.py:
```diff
@@ -0,0 +1,267 @@
+    assert out.size() == (4, 32)
+
+
+def test_static_gcn_conv(device="cpu"):
```
We take the CPU device into account only.

gabyweg76 committed:
remove test functions to put it in the correct folder

gabyweg76 committed:
Test functions for `BayesianGCNConv`

More test functions could be added if necessary

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Adding a docs string

Please tell me if you want to go further in details for more information you can read the following article:
`https://arxiv.org/abs/2012.07923`

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Merge branch 'master' into master

gabyweg76 committed:
Update and rename Bayesian_GCNConv.py to BayesianGCNConv.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 commented:
Here I added some doc string and move the test functions to the tests folder. Probably after that this conv layer is accepted, will be to implement the cugraph package for a better propagation operation. I would be happy to do that as well. Just as long as the theory is fine for this graph convolution layer ..

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+import torch.nn.functional as F
+from torch import Tensor
+from torch.nn import Parameter
+from torch_sparse import SparseTensor
```
```suggestion
```

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.nn.conv.gcn_conv import gcn_norm
+from torch_geometric.typing import Adj, OptPairTensor, OptTensor
```
```suggestion
from torch_geometric.typing import Adj, OptPairTensor, OptTensor, SparseTensor
```

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+
+    def kl_loss(self):
+        sigma_weight = torch.log1p(torch.exp(self.rho_weight))
+        kl = self.kl_div(self.mu_weight, sigma_weight, self.prior_weight_mu,
```
Is it possible to use `torch.nn.functional.kl_div` instead?

gabyweg76 replied:
I would rather keep the kl_div implemented from their version, it seems that they need to use a mu parameter and sigma parameter for the distribution P that we define at the class declaration

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+            (sigma_weight * self.eps_weight.data.normal_())
+
+        if return_kl:
+            kl_weight = self.kl_div(self.mu_weight, sigma_weight,
```
Is that misplaced?

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+                kl = kl_weight + kl_bias
+                return out, kl
+            else:
+                kl = kl_weight
```
```suggestion
```

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+
+    def forward(self, x: Tensor, edge_index: Adj,
+                edge_weight: OptTensor = None,
+                return_kl: bool = True) -> Tensor:
```
```suggestion
                return_kl: bool = True) -> Union[Tensor, Tuple[Tensor, Tensor]]:
```

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+   .. math::
+      \mathbf{X}^{\prime} = \mathbf{X} \mathbf{W}^{T} +  \mathbf{b}
+
+   where :math:\mathbf{W}^{T}  denotes the weights with a fully factorized
```
```suggestion
   where :math:`\mathbf{W}^{T}` denotes the weights with a fully factorized
```

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+   Gaussian distributions represented by :math:`\mu` and :math:`\sigma`, with
+   an unconstrained parameter :math:`\rho` for :math:`\sigma = log(1+exp(\rho))`
+   to avoid non-negative variance.
+   \mathbf{b} denotes the bias which is based on the same structure as the weights
```
```suggestion
   :math:`\mathbf{b}` denotes the bias which is based on the same structure as the weights
```

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,271 @@
+   an unconstrained parameter :math:`\rho` for :math:`\sigma = log(1+exp(\rho))`
+   to avoid non-negative variance.
+   \mathbf{b} denotes the bias which is based on the same structure as the weights
+   CAUTION: the weights and `edge_weight` are different parameters, since the `edge_weight``
```
```suggestion
   CAUTION: the weights and :obj:`edge_weight` are different parameters, since the :obj:`edge_weight``
```

EdisonLeeeee commented on test/nn/conv/BayesianGCNConv_test.py:
```diff
@@ -0,0 +1,341 @@
+from torch_geometric.utils import spmm
+
+
+class BayesianGCNConv(MessagePassing):
```
Let's import it from `torch_geometric.nn`

EdisonLeeeee commented on test/nn/conv/BayesianGCNConv_test.py:
```diff
@@ -0,0 +1,341 @@
+    edge_index = torch.tensor([[0, 0, 0, 1, 2, 3], [1, 2, 3, 0, 0, 0]])
+    conv = BayesianGCNConv(16, 32)
+    out, kl = conv(x, edge_index)
+    print(out.shape, kl.detach())
```
We don't need the print statements. Please use `assert` instead. See also `test/nn/conv/test_gcn_conv.py`.

EdisonLeeeee reviewed (commented):
Thanks for the update. Left some final comments.

gabyweg76 committed:
Update CHANGELOG.md

gabyweg76 committed:
Update torch_geometric/nn/conv/BayesianGCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Merge branch 'master' into master

gabyweg76 committed:
Update torch_geometric/nn/conv/BayesianGCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Update torch_geometric/nn/conv/BayesianGCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Update torch_geometric/nn/conv/BayesianGCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Update torch_geometric/nn/conv/BayesianGCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Update torch_geometric/nn/conv/BayesianGCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Modified the test functions

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Remove return_kl

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

EdisonLeeeee commented on test/nn/conv/BayesianGCNConv_test.py:
```diff
@@ -0,0 +1,128 @@
+import torch
+from torch_sparse import SparseTensor
+
+from torch_geometric.nn.conv.BayesianGCNConv import BayesianGCNConv
```
```suggestion
from torch_geometric.nn import BayesianGCNConv
```
You might need to add `BayesianGCNConv` into `__init__.py` in `conv/`.

EdisonLeeeee commented on torch_geometric/nn/conv/BayesianGCNConv.py:
```diff
@@ -0,0 +1,267 @@
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.nn.conv.gcn_conv import gcn_norm
+from torch_geometric.typing import Adj, OptPairTensor, OptTensor, SparseTensor
+# for the last version of torch_geometric
```
```suggestion
```

gabyweg76 committed:
Update __init__.py

gabyweg76 committed:
Update torch_geometric/nn/conv/BayesianGCNConv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv.py

gabyweg76 committed:
Update BayesianGCNConv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Will modify the test_gcn_conv

SparseTensor is not working for now, i will modify it to make it work

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Create BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 commented:
Test_gcn_conv is now running with only dense matrix for edge_index, if necessary I can implement it later, it worked but only with a previous version of torch_geometric, i think it's depending on the last implementation of `SparseTensor` where gcn_norm is lying upon

gabyweg76 committed:
Update BayesianGCNConv_test.py

Modified test function to add `torch_sparse_coo_tensor`

gabyweg76 committed:
Merge branch 'master' into master

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Include sparse Tensor for BayesianGCNConv

gabyweg76 committed:
Merge branch 'master' into master

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update BayesianGCNConv_test.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 reviewed (commented):
I wrote the sparse tensor implementation its runs well on the torch_geometric version I used but I don't understand why it's not on your current repository.  But the sparse tensor format for the edge_index is not mandatory for my implementation. The main structure remains unchanged

gabyweg76 committed:
Merge branch 'master' into master

EdisonLeeeee committed:
Merge branch 'master' into master

EdisonLeeeee committed:
Update and rename BayesianGCNConv_test.py to test_bayesian_gcn_conv.py

EdisonLeeeee committed:
Update and rename BayesianGCNConv.py to bayesian_gcn_conv.py

EdisonLeeeee committed:
Update __init__.py

EdisonLeeeee committed:
Update bayesian_gcn_conv.py

EdisonLeeeee committed:
Update CHANGELOG.md

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

gabyweg76 committed:
Merge branch 'master' into master

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

gabyweg76 commented:
Last test functions implementation from `test_gcn_conv.py` with SparseTensor.  @EdisonLeeeee, I did your last recommendation, I just skipped the test_gcn_norm_gradient function since the gcn_norm is not modified for this conv layer implementation.. @rusty1s feel free to check the last modification we did together
Looking forward for your feedbacks

EdisonLeeeee commented on torch_geometric/nn/conv/bayesian_gcn_conv.py:
```diff
@@ -0,0 +1,271 @@
+              (sigma_q**2 + (mu_q - mu_p)**2) / (2 * (sigma_p**2)) - 0.5)
+        return kl.mean()
+
+    def kl_loss(self):
```
```suggestion
    def kl_loss(self) -> Tensor:
```

EdisonLeeeee commented on torch_geometric/nn/conv/bayesian_gcn_conv.py:
```diff
@@ -0,0 +1,271 @@
+
+    def forward(
+            self, x: Tensor, edge_index: Adj, edge_weight: OptTensor = None
+    ) -> Union[Tensor, Tuple[Tensor, Tensor]]:
```
```suggestion
    ) -> Tuple[Tensor, Tensor]:
```

EdisonLeeeee commented on test/nn/conv/test_bayesian_gcn_conv.py:
```diff
@@ -0,0 +1,165 @@
+    assert torch.allclose(kl1, kl2, atol=1e-6)
+
+
+def test_gcn_conv():
```
I think the `test_bayesian_conv` test is sufficient, as the others overlap with `GCNConv`.

EdisonLeeeee reviewed (commented):
Left some final reviews. Thank you for all the hard work you've done!

EdisonLeeeee committed:
Merge branch 'master' into master

gabyweg76 committed:
Update torch_geometric/nn/conv/bayesian_gcn_conv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Update torch_geometric/nn/conv/bayesian_gcn_conv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

gabyweg76 committed:
Update test/nn/conv/test_bayesian_gcn_conv.py

Co-authored-by: Jintang Li <cnljt@outlook.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update test_bayesian_conv

i keep only the last function and commented the others, as suggested

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

sorry for all the unnecessary commits

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

EdisonLeeeee committed:
Update bayesian_gcn_conv.py

EdisonLeeeee committed:
Merge branch 'master' into master

EdisonLeeeee committed:
Update test_bayesian_gcn_conv.py

EdisonLeeeee committed:
Update bayesian_gcn_conv.py

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

add few features for testing

gabyweg76 committed:
Merge branch 'master' into master

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gabyweg76 committed:
Update test_bayesian_gcn_conv.py

Corrections

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
